File: app/src/main/java/dev/suhockii/lifetest/util/ui/fragment/InteractionFragment.java
package dev.suhockii.lifetest.util.ui.fragment;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.LayoutRes;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.arellomobile.mvp.MvpAppCompatFragment;

import dev.suhockii.lifetest.di.Injectable;

/**
 * Sends signal to activity that implements {@link OnFragmentInteractionListener#onFragmentInteraction}
 * that fragment has been connected.
 */
public abstract class InteractionFragment extends MvpAppCompatFragment implements Injectable {
    protected OnFragmentInteractionListener onFragmentInteractionListener;

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        onFragmentInteractionListener.onFragmentInteraction(getToolbarTitle());
        return inflater.inflate(getLayoutRes(), container, false);
    }

    @NonNull
    protected abstract String getToolbarTitle();

    @LayoutRes
    protected abstract int getLayoutRes();

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        try {
            onFragmentInteractionListener = (OnFragmentInteractionListener) getActivity();
        } catch (ClassCastException e) {
            throw new ClassCastException(getActivity().toString()
                    + " must implement OnFragmentInteractionListener");
        }
    }

    @Override
    public void onDetach() {
        super.onDetach();
        onFragmentInteractionListener = null;
    }

    public interface OnFragmentInteractionListener {

        /**
         * Call when fragment attaches to activity
         */
        void onFragmentInteraction(String title);
    }
}

File: app/src/main/java/dev/suhockii/lifetest/util/ui/fragment/InteractionFragment.kt
package dev.suhockii.lifetest.util.ui.fragment

import android.content.Context
import android.os.Bundle
import androidx.annotation.LayoutRes
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import moxy.MvpAppCompatFragment

import dev.suhockii.lifetest.di.Injectable

/**
 * Sends signal to activity that implements [OnFragmentInteractionListener.onFragmentInteraction]
 * that fragment has been connected.
 */
abstract class InteractionFragment : MvpAppCompatFragment(), Injectable {
    private var onFragmentInteractionListener: OnFragmentInteractionListener? = null

    protected abstract val toolbarTitle: String

    @get:LayoutRes
    protected abstract val layoutRes: Int

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        onFragmentInteractionListener!!.onFragmentInteraction(toolbarTitle)
        return inflater.inflate(layoutRes, container, false)
    }

    override fun onAttach(context: Context?) {
        super.onAttach(context)
        try {
            onFragmentInteractionListener = activity as OnFragmentInteractionListener?
        } catch (e: ClassCastException) {
            throw ClassCastException(activity!!.toString() + " must implement OnFragmentInteractionListener")
        }

    }

    override fun onDetach() {
        super.onDetach()
        onFragmentInteractionListener = null
    }

    interface OnFragmentInteractionListener {

        /**
         * Call when fragment attaches to activity
         */
        fun onFragmentInteraction(title: String)
    }
}
